docs/source/migration-guides/post-0.8-deserialization.md

```diff
@@ -0,0 +1,264 @@
+
+The legacy API works by deserializing rows in the query response to a sequence of `Row`s. The `Row` is just a `Vec<Option<CqlValue>>`, where `CqlValue` is an enum that is able to represent any CQL value.
+
+The user can request this type-erased representation to be converted into something useful. There are two traits that power this:
+
+__`FromRow`__
+
+```rust
+# extern crate scylla;
+# use scylla::frame::response::cql_to_rust::FromRowError;
+# use scylla::frame::response::result::Row;
+pub trait FromRow: Sized {
+    fn from_row(row: Row) -> Result<Self, FromRowError>;
+}
+```
+
+__`FromCqlVal`__
+
+```rust
+# extern crate scylla;
+# use scylla::frame::response::cql_to_rust::FromCqlValError;
+// The `T` parameter is supposed to be either `CqlValue` or `Option<CqlVal>`
+pub trait FromCqlVal<T>: Sized {
+    fn from_cql(cql_val: T) -> Result<Self, FromCqlValError>;
+}
+```
+
+These traits are implemented for some common types:
+
+- `FromRow` is implemented for tuples up to 16 elements,
+- `FromCqlVal` is implemented for a bunch of types, and each CQL type can be converted to one of them.
+
+While it's possible to implement those manually, the driver provides procedural macros for automatic derivation in some cases:
+
+- `FromRow` - implements `FromRow` for a struct.
+- `FromUserType` - generated an implementation of `FromCqlVal` for the struct, trying to parse the CQL value as a UDT.
+
+Note: the macros above have a default behavior that is different than what `FromRow` and `FromUserTypes` do.
+
+### New traits
+
+The new API introduce two analogous traits that, instead of consuming pre-parsed `Vec<Option<CqlValue>>`, are given raw, serialized data with full information about its type. This leads to better performance and allows for better type safety.
+
+The new traits are:
+
+__`DeserializeRow<'frame>`__
+
+```rust
+# extern crate scylla;
+# use scylla::types::deserialize::row::ColumnIterator;
+# use scylla::frame::frame_errors::ParseError;
+# use scylla::frame::response::result::ColumnSpec;
+pub trait DeserializeRow<'frame>
+where
+    Self: Sized,
+{
+    fn type_check(specs: &[ColumnSpec]) -> Result<(), ParseError>;
+    fn deserialize(row: ColumnIterator<'frame>) -> Result<Self, ParseError>;
+}
+```
+
+__`DeserializeCql<'frame>`__
+
+```rust
+# extern crate scylla;
+# use scylla::types::deserialize::row::ColumnIterator;
+# use scylla::types::deserialize::FrameSlice;
+# use scylla::frame::frame_errors::ParseError;
+# use scylla::frame::response::result::ColumnType;
+pub trait DeserializeCql<'frame>
+where
+    Self: Sized,
+{
+    fn type_check(typ: &ColumnType) -> Result<(), ParseError>;
+    fn deserialize(
+        typ: &'frame ColumnType,
+        v: Option<FrameSlice<'frame>>,
+    ) -> Result<Self, ParseError>;
+}
+```
+
+The above traits have been implemented for the same set of types as `FromRow` and `FromCqlVal`, respectively. Notably, `DeserializeRow` is implemented for `Row`, and `DeserializeCql` is implemented for `CqlValue`.
```

Comment:
It can be helpful to remind that `Row` and `CqlValue` are representatives of the legacy API.